matrix_sdk/src/request_builder.rs

```diff
@@ -281,11 +282,11 @@ impl MessagesRequestBuilder {
 impl Into<get_message_events::Request> for MessagesRequestBuilder {
     fn into(self) -> get_message_events::Request {
         get_message_events::Request {
-            room_id: self.room_id.expect("`room_id` and `from` need to be set"),
-            from: self.from.expect("`room_id` and `from` need to be set"),
+            room_id: self.room_id,
+            from: self.from,
             to: self.to,
             dir: self.direction.unwrap_or(Direction::Backward),
-            limit: self.limit,
+            limit: self.limit.map_or(Ok(uint!(10)), UInt::try_from).ok(),
```

Comment:
`limit: self.limit.map_or(Some(uint!(10)), |n| Some(n.into()))` is this what you were thinking or this reads better
```
limit: self
    .limit
    .and_then(|n| Some(n.into()))
    .unwrap_or(Some(uint!(10))),
```

Reply:
No, just a `limit: self.limit.map(UInt::from)`.

Reply:
The RoomListFilterBuilder yes, that doesn't need a default, I'm changing that. `get_message_event::Request`s limit field specifies a default of 10.

Reply:
That [request](https://docs.rs/ruma-client-api/0.9.0/ruma_client_api/r0/message/get_message_events/struct.Request.html#structfield.limit) also takes an `Option<Uint>`, the default of 10 is on the server if no limit is given. Always sending a default limit that is equivalent to the server default limit seems like a waste, no?

Even so I'm not so much complaining about setting a default value, I'm complaining about the usage of `try_from()` which returns a `Result` when we can use `from()` since the conversion will never fail.

Reply:
Ohhh duh not our problem as a client yeeeesh sory about that :facepalm: